src/reconcile/rec_write.c

```diff
@@ -1260,6 +1260,10 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins,
 		if (WT_TXNID_LT(max_txn, txnid))
 			max_txn = txnid;
 
+		/* Track the first update with non-zero timestamp. */
+		if (first_ts_upd == NULL && upd->start_ts != WT_TS_NONE)
+			first_ts_upd = upd;
```

Comment:
This is the main fix here: include the timestamp from prepapred updates in the calculated `unstable_timestamp`.